### PR TITLE
its espresso_event_categories not espresso_event_category

### PR DESCRIPTION
**Pull request by mnelson4**
Addresses https://github.com/eventespresso/event-espresso-core/issues/417

### CODE REVIEW

**Comment by mnelson4** [comment]
## Testing
* [ ] verify this resolves https://github.com/eventespresso/event-espresso-core/issues/417, although I'm not sure where the affected code was actually in use in EE core (so this was probably an issue only for custom code that made use of these helper functions)

**Comment by joshfeck** [comment]
If this was probably an issue only for custom code that made use of these helper functions, then there's not much that can be done with regards to testing, correct?

**Comment by youri--** [comment]
As for the custom code... Issue #431 describes a custom code example of how this bug came to light (I reported it on the EE forum). I.e. using PHP to programmatically add a event_espresso_category to an event.

**Comment by mnelson4** [comment]
Merged BUG/invalid-field-count into this because otherwise we couldn't use the method `EE_CPT_Base::add_event_category()`

**Comment by mnelson4** [comment]
Passing for review again

File: tests/testcases/core/db_classes/EE_CPT_Base_Test.php [hunk 1]
```diff
@@ -118,6 +118,25 @@ public function testGetPrettyCurrentPostRemainingTheSame()
...
+        /**
+         * @var $e EE_Event
+         */
+        $ee_term_taxonomy = $e->add_event_category(
+            'Rebel Ships',
+            'Events look at important Rebel Space Ships'
+        );
+        $this->assertTrue($ee_term_taxonomy instanceof EE_Term_Taxonomy);
+        $this->assertNotEquals(0, $ee_term_taxonomy);
+        $this->assertEquals('espresso_event_categories', $ee_term_taxonomy->taxonomy());
```
Comment (nerrad) [inline tests/testcases/core/db_classes/EE_CPT_Base_Test.php]: as mentioned in my review in the other ticket, I don't think the second assertion is necessary.

Comment (mnelson4) [reply tests/testcases/core/db_classes/EE_CPT_Base_Test.php]: ah good points, I meant to do the first one but got dizzied by the rest of this UI. The second one I hadn't thought of but makes a ton of sense

**Comment by mnelson4** [comment]
this will be ready ot merge once the other passes review and I resolve the conflict

**Comment by mnelson4** [comment]
huh, because this was merged into BUG/invalid-field-count AFTER it had already been merged into master, I don't think these changes are in master

**Comment by mnelson4** [comment]
yep, this isn't yet in master. For example, `EEM_CPT_Base::EVENT_CATEGORY_TAXONOMY` is still `espresso_event_category` on master branch, but this branch changed that

**Pull request merged**